linkcheck/logger/json.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (C) 2022 Mark Ferrell
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+JSON logger.
+"""
+
+import json
+from . import _Logger
+
+
+class JSONLogger(_Logger):
+    """JSON logger; easy to parse with jq. """
+
+    LoggerName = 'json'
+    LoggerArgs = {
+        "filename": "linkchecker-out.json",
+        "indent": "",
```

Comment:
Looks like the intent is 4 is the default in [linkcheckerrc](https://github.com/linkchecker/linkchecker/pull/618/files#diff-da6a66e321c223b0ae2bd60cf52ae02c6f6f1fd3b553282dae197c5ccccb79ae).